modules/menu/menu.c

```diff
@@ -1159,6 +1159,68 @@ int menu_param_decode(const char *prm, const char *name, struct pl *val)
 }
 
 
+/**
+ * Find ua and call from command arguments.
+ *
+ * Assumes that the first argument passed in carg->prm is a valid call-id
+ * (if passed at all).
+ *
+ * @param carg   Command arguments
+ * @param uap    Pointer-pointer to ua. Set on successful return.
+ * @param callp  Pointer-pointer to call. Set on successful return.
+ *
+ * @return 0 for success, otherwise errorcode
+ */
+int menu_get_call_ua(const struct cmd_arg *carg, struct ua **uap,
+		     struct call **callp)
+{
+	int err = 0;
+	struct ua *ua;
+	struct call *call;
+	const char *eq;
+	char *cid = NULL;
+	struct pl pl = PL_INIT;
+
+	if (!carg || !uap || !callp)
+		return EINVAL;
+
+	ua = carg->data ? carg->data : menu_uacur();
+	call = ua_call(ua);
+
+	(void)re_regex(carg->prm, str_len(carg->prm), "[^ ]+", &pl);
```

Comment:
Is it possible to combine the two checks with regexp `"[^ =]+"`?

Reply:
I tried, but was not successful. With your mentioned regex, the Call-Id `abcd=ef` would be matched as `abcd` and we would look for the Call-Id `abcd`. Instead we want to detect that this is not a valid Call-Id.